### PR TITLE
Remove remaining uses of boost.

### LINKED ISSUE #150: laser_filters in rolling fails to build on rhel targets
The package laser_filters in rolling has been detected as not building on all platforms in the buildfile rhel. The release manager for rolling will consider disabling this build if it continues to fail to build.
- jenkins_urls:
  - [rhel 8 x86_64](https://build.ros2.org/view/Rbin_rhel_el864/job/Rbin_rhel_el864__laser_filters__rhel_8_x86_64__binary)
This is being filed because this package is about to be blacklisted. If this ticket is resolved please review whether it can be removed from the blacklist that should cross reference here.

```
DEBUG: [ 71%] Linking CXX executable scan_to_cloud_filter_chain
DEBUG: /usr/bin/cmake -E cmake_link_script CMakeFiles/scan_to_cloud_filter_chain.dir/link.txt --verbose=1
DEBUG: /usr/bin/c++ -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -Wl,-z,relro  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld CMakeFiles/scan_to_cloud_filter_chain.dir/src/scan_to_cloud_filter_chain.cpp.o -o scan_to_cloud_filter_chain  -Wl,-rpath,/builddir/build/BUILD/ros-rolling-laser-filters-2.0.1/obj-x86_64-redhat-linux-gnu:/opt/ros/rolling/lib:/opt/ros/rolling/lib64: liblaser_scan_filters.so /opt/ros/rolling/lib/libstatic_transform_broadcaster_node.so /opt/ros/rolling/lib/liblaser_geometry.so /opt/ros/rolling/lib/librclcpp_lifecycle.so /usr/lib64/libtinyxml2.so /opt/ros/rolling/lib/libtf2_ros.so /opt/ros/rolling/lib/libmessage_filters.so /opt/ros/rolling/lib/librclcpp_action.so /opt/ros/rolling/lib/librcl_action.so /opt/ros/rolling/lib/libtf2_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libtf2_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libtf2_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libtf2_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libtf2_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libtf2_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/libaction_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libaction_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libaction_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libaction_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libaction_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libaction_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/libunique_identifier_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libunique_identifier_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libunique_identifier_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libunique_identifier_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libunique_identifier_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libunique_identifier_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/libcomponent_manager.so /opt/ros/rolling/lib/libclass_loader.so /opt/ros/rolling/lib/libcomposition_interfaces__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libcomposition_interfaces__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libcomposition_interfaces__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libcomposition_interfaces__rosidl_generator_py.so /opt/ros/rolling/lib/libcomposition_interfaces__rosidl_generator_c.so /opt/ros/rolling/lib/libcomposition_interfaces__rosidl_typesupport_c.so /opt/ros/rolling/lib/libsensor_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libsensor_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libsensor_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libsensor_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libsensor_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libsensor_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/libtf2.so /opt/ros/rolling/lib/libgeometry_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libgeometry_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libgeometry_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libgeometry_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libgeometry_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libgeometry_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib64/libconsole_bridge.so.1.0 /opt/ros/rolling/lib/librclcpp.so -lpthread /opt/ros/rolling/lib/liblibstatistics_collector.so /opt/ros/rolling/lib/liblibstatistics_collector_test_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/liblibstatistics_collector_test_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/liblibstatistics_collector_test_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/liblibstatistics_collector_test_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/liblibstatistics_collector_test_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/liblibstatistics_collector_test_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/libstd_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libstd_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libstd_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libstd_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libstd_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libstd_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/librosgraph_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/librosgraph_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/librosgraph_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/librosgraph_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/librosgraph_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/librosgraph_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/libstatistics_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libstatistics_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libstatistics_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libstatistics_msgs__rosidl_generator_py.so /opt/ros/rolling/lib/libstatistics_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/libstatistics_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/librcl_lifecycle.so /opt/ros/rolling/lib/librcl.so /opt/ros/rolling/lib/librcl_yaml_param_parser.so /opt/ros/rolling/lib/libyaml.so /opt/ros/rolling/lib/librcl_interfaces__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/librcl_interfaces__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/librcl_interfaces__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/librcl_interfaces__rosidl_generator_py.so /opt/ros/rolling/lib/librcl_interfaces__rosidl_generator_c.so /opt/ros/rolling/lib/librcl_interfaces__rosidl_typesupport_c.so /opt/ros/rolling/lib/libbuiltin_interfaces__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/libbuiltin_interfaces__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/libbuiltin_interfaces__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/libbuiltin_interfaces__rosidl_generator_py.so /opt/ros/rolling/lib/libbuiltin_interfaces__rosidl_generator_c.so /opt/ros/rolling/lib/libbuiltin_interfaces__rosidl_typesupport_c.so /opt/ros/rolling/lib/librmw_implementation.so /opt/ros/rolling/lib/libament_index_cpp.so /opt/ros/rolling/lib/librmw.so /opt/ros/rolling/lib/librcl_logging_spdlog.so /opt/ros/rolling/lib/librcl_logging_interface.so /opt/ros/rolling/lib/libtracetools.so /opt/ros/rolling/lib/liblifecycle_msgs__rosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/liblifecycle_msgs__rosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/librosidl_typesupport_introspection_cpp.so /opt/ros/rolling/lib/librosidl_typesupport_introspection_c.so /opt/ros/rolling/lib/liblifecycle_msgs__rosidl_typesupport_cpp.so /opt/ros/rolling/lib/librosidl_typesupport_cpp.so /opt/ros/rolling/lib/liblifecycle_msgs__rosidl_generator_py.so /usr/lib64/libpython3.6m.so /opt/ros/rolling/lib/liblifecycle_msgs__rosidl_generator_c.so /opt/ros/rolling/lib/liblifecycle_msgs__rosidl_typesupport_c.so /opt/ros/rolling/lib/librosidl_typesupport_c.so /opt/ros/rolling/lib/librcpputils.so /opt/ros/rolling/lib/librosidl_runtime_c.so /opt/ros/rolling/lib/librcutils.so -ldl
DEBUG: liblaser_scan_filters.so: undefined reference to `boost::system::generic_category()'
DEBUG: collect2: error: ld returned 1 exit status
DEBUG: make[2]: Leaving directory '/builddir/build/BUILD/ros-rolling-laser-filters-2.0.1/obj-x86_64-redhat-linux-gnu'
DEBUG: make[1]: Leaving directory '/builddir/build/BUILD/ros-rolling-laser-filters-2.0.1/obj-x86_64-redhat-linux-gnu'
DEBUG: make[2]: *** [CMakeFiles/scan_to_cloud_filter_chain.dir/build.make:211: scan_to_cloud_filter_chain] Error 1
DEBUG: make[1]: *** [CMakeFiles/Makefile2:219: CMakeFiles/scan_to_cloud_filter_chain.dir/all] Error 2
DEBUG: make: *** [Makefile:149: all] Error 2
```

### PR DESCRIPTION
**Pull request by clalancette**
All of that functionality is now available in std:: .  Also, this
should fix the build on RHEL.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

This should fix #150

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -13,17 +13,17 @@
...
 
+  <depend>angles</depend>
+  <depend>filters</depend>
+  <depend>laser_geometry</depend>
+  <depend>message_filters</depend>
   <depend>pluginlib</depend>
-  <depend>sensor_msgs</depend>
   <depend>rclcpp</depend>
+  <depend>rclcpp_lifecycle</depend>
+  <depend>sensor_msgs</depend>
```
Comment (jonbinney) [inline package.xml:23]: Looks like you also ran a formatter on this?

**Review by jonbinney — APPROVED** [review]
LGTM

**Comment by jonbinney** [comment]
Thanks @clalancette !

File: package.xml (hunk 1 above)
Comment (clalancette) [reply package.xml:23]: Yeah, that's correct.  While I was in here I just decided to alphabetize.

**Comment by clalancette** [comment]
> Thanks @clalancette !

Thanks for reviewing and merging.

When you get a chance, a new release of this package into Rolling and Humble would be most appreciated.  Thanks.

**Comment by jonbinney** [comment]
@clalancette done.
- Release in rolling: https://github.com/ros/rosdistro/pull/33390
- in humble: https://github.com/ros/rosdistro/pull/33391

**Comment by clalancette** [comment]
Thank you so much!

**Pull request merged**